internal/daemon/daemon.go

```diff
@@ -206,16 +215,18 @@ func (d *Daemon) init(listenPort string, schemaExtensions []schema.Update, apiEx
 
 	if listenPort != "" {
 		serverEndpoints = []rest.Resources{resources.PublicEndpoints}
-		serverEndpoints = append(serverEndpoints, coreEndpoints...)
-		server := d.initServer(serverEndpoints...)
-		url := api.NewURL().Host(fmt.Sprintf(":%s", listenPort))
-		network := endpoints.NewNetwork(d.shutdownCtx, endpoints.EndpointNetwork, server, *url, d.serverCert)
-		err = d.endpoints.Add(network)
+		err = d.addCoreServers(true, *listenAddr, d.ServerCert(), serverEndpoints)
 		if err != nil {
 			return err
 		}
 	}
 
+	// Add extension servers before post-join hook.
+	err = d.addExtensionServers(true, d.ServerCert(), listenAddr.URL.Host)
```

Comment:
Do we need this here? Is the assumption here that we may have project defined network listeners that would be fully configured prior to initialisation?